app/controllers/api_controller/collection_config.rb

```diff
@@ -0,0 +1,110 @@
+class ApiController
+  class CollectionConfig < Config::Options
+    def collection?(collection_name)
```

Comment:
I agree. Further abstraction for single collection would be great. However, I would prefer to leave this for a separate pr.